src/java/util/ImageUtils.java

```diff
@@ -45,4 +49,70 @@ public static BufferedImage scaleImage(BufferedImage image, int width, int heigh
+                byte greenByte = (byte) ((Float) (greenMask.getValueAt(x, y) * scaleMultiplier)).intValue();
+                byte blueByte = (byte) ((Float) (blueMask.getValueAt(x, y) * scaleMultiplier)).intValue();
+                index += 3;
+                byteArray[index - 2] = redByte;
+                byteArray[index - 1] = greenByte;
+                byteArray[index] = blueByte;
+            }
+        }
+        DataBuffer buffer = new DataBufferByte(byteArray, byteArray.length);
+        WritableRaster raster = Raster.createInterleavedRaster(buffer, size, size, 3 * size, 3, new int[]{0, 1, 2}, new Point(0, 0));
+        ColorModel colorModel = new ComponentColorModel(ColorModel.getRGBdefault().getColorSpace(), false, true, Transparency.TRANSLUCENT, DataBuffer.TYPE_BYTE);
+        RenderedImage image = (RenderedImage) new BufferedImage(colorModel, raster, true, null);
+        ImageIO.write(image, "png", new File(path));
+        System.out.println("PNG created at " + path);
+    }
+
+    public static void writeAutomaticallyScaledPNGFromMasks(FloatMask redMask, FloatMask greenMask, FloatMask blueMask, String path) throws IOException {
+        int size = redMask.getSize();
+        if (size != greenMask.getSize() || size != blueMask.getSize()) {
+            throw new IllegalArgumentException("Masks not the same size: redMask is " + redMask.getSize() +
+                    ", greenMask is " + greenMask.getSize() + " and blueMask is " + blueMask.getSize());
+        }
+        float scaleMultiplier = 255 / StrictMath.max(StrictMath.max(redMask.getMax(), greenMask.getMax()), blueMask.getMax());
+        int length = size * size * 3;
+        int index = -1;
+        final byte[] byteArray = new byte[length];
+        for (int x = 0; x < size; x++) {
+            for (int y = 0; y < size; y++) {
+                byte redByte = (byte) ((Float) (redMask.getValueAt(x, y) * scaleMultiplier)).intValue();
+                byte greenByte = (byte) ((Float) (greenMask.getValueAt(x, y) * scaleMultiplier)).intValue();
+                byte blueByte = (byte) ((Float) (blueMask.getValueAt(x, y) * scaleMultiplier)).intValue();
+                index += 3;
+                byteArray[index - 2] = redByte;
+                byteArray[index - 1] = greenByte;
+                byteArray[index] = blueByte;
+            }
+        }
+        DataBuffer buffer = new DataBufferByte(byteArray, byteArray.length);
+        WritableRaster raster = Raster.createInterleavedRaster(buffer, size, size, 3 * size, 3, new int[]{0, 1, 2}, new Point(0, 0));
+        ColorModel colorModel = new ComponentColorModel(ColorModel.getRGBdefault().getColorSpace(), false, true, Transparency.TRANSLUCENT, DataBuffer.TYPE_BYTE);
+        RenderedImage image = (RenderedImage) new BufferedImage(colorModel, raster, true, null);
+        ImageIO.write(image, "png", new File(path));
+        System.out.println("PNG created at " + path);
```

Comment:
This should just call writePNGfromMasks(redMask, blueMask, greenMask, scaleMultipler, path) to reduce code duplication